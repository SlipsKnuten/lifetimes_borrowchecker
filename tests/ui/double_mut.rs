fn main() {
    let mut s = String::from("hi");
    let m1 = &mut s;
    let m2 = &mut s;
    m1.push('!');
    m2.push('?');
}
