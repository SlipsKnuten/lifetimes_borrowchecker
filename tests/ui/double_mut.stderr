error[E0499]: cannot borrow `s` as mutable more than once at a time
 --> tests/ui/double_mut.rs:4:14
  |
3 |     let m1 = &mut s;
  |              ------ first mutable borrow occurs here
4 |     let m2 = &mut s;
  |              ^^^^^^ second mutable borrow occurs here
5 |     m1.push('!');
  |     -- first borrow later used here
